queries/nix/injections.scm

```diff
@@ -1 +1,98 @@
+  ]
+  (#match? @_func "(^|\\.)match$"))
+
+(binding
+  attrpath: (attrpath (identifier) @_path)
+  expression: [
+    (string_expression (string_fragment) @bash)
+    (indented_string_expression (string_fragment) @bash)
+  ]
+  (#match? @_path "(^\\w*Phase|(pre|post)\\w*|(.*\\.)?\\w*([sS]cript|[hH]ook)|(.*\\.)?startup)$"))
+
+(apply_expression
+  function: (_) @_func
+  argument: (_ (_)* (_ (_)* (binding
+    attrpath: (attrpath (identifier) @_path)
+    expression: [
+      (string_expression (string_fragment) @bash)
+      (indented_string_expression (string_fragment) @bash)
+    ])))
+  (#match? @_func "(^|\\.)writeShellApplication$")
+  (#match? @_path "^text$"))
+
+(apply_expression
+  (apply_expression
+    function: (apply_expression function: (_) @_func)
+    argument: [
+      (string_expression (string_fragment) @bash)
+      (indented_string_expression (string_fragment) @bash)
+    ])
+  (#match? @_func "(^|\\.)runCommand(((No)?CC)?(Local)?)?$"))
+
+((apply_expression
+  function: (apply_expression function: (_) @_func)
+  argument: [
+    (string_expression (string_fragment) @bash)
+    (indented_string_expression (string_fragment) @bash)
+  ])
+  (#match? @_func "(^|\\.)write(Bash|Dash|ShellScript)(Bin)?$"))
+
+((apply_expression
+  function: (apply_expression function: (_) @_func)
+  argument: [
+    (string_expression (string_fragment) @fish)
+    (indented_string_expression (string_fragment) @fish)
+  ])
+  (#match? @_func "(^|\\.)writeFish(Bin)?$"))
+
+((apply_expression
+  function: (apply_expression
+    function: (apply_expression function: (_) @_func))
+  argument: [
+    (string_expression (string_fragment) @haskell)
+    (indented_string_expression (string_fragment) @haskell)
+  ])
+  (#match? @_func "(^|\\.)writeHaskell(Bin)?$"))
+
+((apply_expression
+  function: (apply_expression
+    function: (apply_expression function: (_) @_func))
+  argument: [
+    (string_expression (string_fragment) @javascript)
+    (indented_string_expression (string_fragment) @javascript)
+  ])
+  (#match? @_func "(^|\\.)writeJS(Bin)?$"))
+
+((apply_expression
+  function: (apply_expression
+    function: (apply_expression function: (_) @_func))
+  argument: [
+    (string_expression (string_fragment) @perl)
+    (indented_string_expression (string_fragment) @perl)
+  ])
+  (#match? @_func "(^|\\.)writePerl(Bin)?$"))
+
+((apply_expression
+  function: (apply_expression
+    function: (apply_expression function: (_) @_func))
+  argument: [
+    (string_expression (string_fragment) @python)
+    (indented_string_expression (string_fragment) @python)
+  ])
+  (#match? @_func "(^|\\.)write(PyPy|Python)[23](Bin)?$"))
+
+((apply_expression
+  function: (apply_expression
+    function: (apply_expression function: (_) @_func))
+  argument: [
+    (string_expression (string_fragment) @rust)
+    (indented_string_expression (string_fragment) @rust)
+  ])
+  (#match? @_func "(^|\\.)writeRust(Bin)?$"))
```

Comment:
have you tested whether those injection queries have any performance implications?

Reply:
i seems to have slowed down a little bit on a file with a bunch of injections, on a large (30k+ lines) file with little to none injections, it seems to have not much of a difference

is there anyway to make this faster? e.g. combining the writers since theres a lot of duplicated code. I'm not very familiar with the query syntax

Reply:
I it's probably our task to profile the queries more to find whether we have any regressions on standardized tests and also to make injections less expensive. I think we can merge. We will see whether it caused any problems